modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -67,7 +68,7 @@ public JavaClientCodegen() {
         cliOptions.add(CliOption.newBoolean(USE_RX_JAVA2, "Whether to use the RxJava2 adapter with the retrofit2 library."));
         cliOptions.add(CliOption.newBoolean(PARCELABLE_MODEL, "Whether to generate models for Android that implement Parcelable with the okhttp-gson library."));
         cliOptions.add(CliOption.newBoolean(USE_PLAY_WS, "Use Play! Async HTTP client (Play WS API)"));
-        cliOptions.add(CliOption.newString(PLAY_VERSION, "Version of Play! Framework (possible values \"play24\", \"play25\")"));
```

Comment:
Do we really have to support each and every version of Play ? This is getting really messy...

Reply:
Since each change of version of play means that everything changes completly, it is so far necessary to keep all of them...

Reply:
This is a retrofit generator, not a Play one. Also it seems to me that a lot of templates are just duplicated between 2.5 and 2.6

Reply:
I am glad that the client generator for play is using retrofit. 

There is a folder called common where all duplicated templates between versions 2.5 and 2.6 should be. If there is a duplicated template not located there, this PR should be rejected

Reply:
I looked and it seems 2.6 is almost entirely duplicate of 2.5 (minor changes but that probably also work for 2.5)
I also think it should be possible to have common templates between 2.4 and 2.5 with minimal work which would greatly simplify the Java generator code.
And 2.5 seems to introduce adapters for CompletionStage (CompletableFuture) which are already part of retrofit-adapters so they could probably be removed.

Reply:
Those minor changes don't work on play 2.5 (100% sure)

Reply:
Eventually, wouldn't Play developers prefer to get scala.concurrent.Future rather than CompletionStage/CompletableFuture ?

Reply:
> Those minor changes don't work on play 2.5 (100% sure)

OK so they should be applied by mustache tags, not by duplicating the templates.

Reply:
Meanwhile there are 2 different versions of Play (Scala and Java), we should have an autogenerated client for both versions. 

I don't know what could be the reason to program play with java. Moreover, I don't know the reason to program on play at all 🍡

Reply:
ok, I have merged the play version templates

Reply:
:+1: great work !